deepchecks/nlp/utils/embeddings_display.py

```diff
@@ -0,0 +1,163 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Utils module for calculating embeddings display checks."""
+
+import os
+import time
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from PyNomaly import loop
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.metrics import log_loss, roc_auc_score
+from sklearn.model_selection import train_test_split
+
+from deepchecks.core.check_utils.multivariate_drift_utils import auc_to_drift_score, build_drift_plot
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.nlp.utils.embeddings_calculator import clean_special_chars
+from deepchecks.utils.dataframes import floatify_dataframe
+
+
+def create_performance_display(text: pd.Series, embeddings: np.ndarray, proba: np.ndarray, y_true: np.ndarray,
+                                 labels: List[str], path: str, sample_size: int = 10000, verbose: bool = False,
+                                 indexes_to_highlight: Dict[str, List[int]] = None):
+    if list(sorted(labels)) != list(labels):
+        raise DeepchecksValueError('Labels must be sorted in an alphanumeric order')
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+    loss_per_sample = [log_loss([y_true], [y_proba], labels=sorted(labels)) for
+                       y_true, y_proba in zip(y_true, proba)]
+
+    text_dataframe = pd.DataFrame(text)
+    text_dataframe.columns = ['text']
+    text_dataframe['loss per sample'] = loss_per_sample
```

Comment:
Didn't see that! How does that look in the display?

Reply:
not amazing not bad